trl/trainer/reward_trainer.py

```diff
@@ -145,8 +145,25 @@ def __init__(
                 "PEFT is not installed and you passed a `peft_config` in the trainer's kwargs, please install it to use the PEFT models"
             )
         elif is_peft_available() and peft_config is not None:
+            if not isinstance(peft_config, PeftConfig):
+                raise ValueError(
+                    "If you want to use the PeftModel, you need to pass a valid PeftConfig object to the RewardTrainer."
+                    f" and you passed a {type(peft_config)}."
+                )
+                
```

Comment:
```suggestion
```

I don't think it's necessary here. The code will eventually fail if the peft config doesn't have the right type